### ISSUE TITLE
URLSessionConfiguration timeoutIntervalForRequest not supported

### ISSUE THREAD

**Issue by CalvHobbes** [AliSoftware/OHHTTPStubs#229]
I've noticed that if I create a new sessionconfiguration with a specific timeoutIntervalForRequest, the request does not time out when I use OHHTTPStubs:

 let config = URLSessionConfiguration.default
  config.timeoutIntervalForRequest = 2

  stub(condition: isHost("httpbin.org") && isPath("/status/200"))
        { _ in
            let stubData = "hello".data(using: String.Encoding.utf8)
            return OHHTTPStubsResponse(data: stubData!, statusCode:200, headers:nil)
                .responseTime(4)
        }

When I make a request using the above session configuration, it should timeout in 2 seconds but it does not.  It also does not work if I use the timeoutInterval on URLRequest.

However, if I set the session configuration's timeoutIntervalForResource value, the timeout happens as expected.

I'm using Alamofire to make the requests by the way. Is this a known issue?

**Comment by AliSoftware** [comment]
I have to admit I never tested that. Will have to investigate to know if it's an existing bug of OHHTTPStubs itself, of Alamofire, or if it's even possible at all to intercept that when using a custom `NSURLProtocol` (like this pod does to intercept requests)…

**Comment by CalvHobbes** [comment]
I used a custom UrlProtocol to get around this issue. I set this in the .protocolClasses property of my sessionconfiguration. The custom protocol basically uses an internal session to forward the request  (which incidentally is to my  OHHTTPStubs stub ).

**Comment by AliSoftware** [comment]
Hey @CalvHobbes 

Thanks for your feedback.

If you managed to make it work, I'd be very interested to see your solution.
You speak about using a custom `NSURLProtocol` and setting it in the `protocolClasses` property, but basically that's also what `OHHTTPStubs` does internally, so I'm intrigued.

Maybe your workaround could be integrated as a fix in `OHHTTPStubs` itself (feel free to do a PR) so others can profit?

**Comment by CalvHobbes** [comment]
Hello, I took another look at this. Turns out, the framework is fine. I was using .responseTime instead of .requestTime to delay the return of the response from the stub, which is why it was returning the response earlier than expected (and my request was therefore not timing out) :) Think we're ok wiht the current code. FYI though, for the custom UrlProtocol, I just used the standard code from one of Alamofire test UrlProtocol classes.
